presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -775,6 +795,64 @@ private void queryCompletedEvent(Optional<ExecutionFailureInfo> failureInfo)
             queryInfoOutputPath.ifPresent(path -> writeQueryInfo(path, queryInfo, queryInfoJsonCodec));
         }
 
+        private void processShuffleStats()
+        {
+            List<PrestoSparkShuffleStats> statsList = new ArrayList<>(shuffleStatsCollector.value());
+            Map<ShuffleStatsKey, List<PrestoSparkShuffleStats>> statsMap = new HashMap<>();
+            Set<Integer> fragments = new HashSet<>();
+            for (PrestoSparkShuffleStats stats : statsList) {
+                ShuffleStatsKey key = new ShuffleStatsKey(stats.getFragmentId(), stats.getOperation());
+                statsMap.computeIfAbsent(key, (ignored) -> new ArrayList<>()).add(stats);
+                fragments.add(stats.getFragmentId());
+            }
+            List<Integer> sortedFragments = new ArrayList<>(fragments);
+            sortedFragments.sort(null);
+            log.info("Shuffle statistics summary:");
+            for (Integer fragment : sortedFragments) {
+                List<PrestoSparkShuffleStats> readStatistics = statsMap.get(new ShuffleStatsKey(fragment, READ));
+                if (readStatistics != null) {
+                    logShuffleStatsSummary(fragment, READ, readStatistics);
+                }
+                List<PrestoSparkShuffleStats> writeStatistics = statsMap.get(new ShuffleStatsKey(fragment, WRITE));
+                if (writeStatistics != null) {
+                    logShuffleStatsSummary(fragment, WRITE, writeStatistics);
+                }
+            }
+        }
+
+        private void logShuffleStatsSummary(int fragmentId, Operation operation, List<PrestoSparkShuffleStats> statsList)
+        {
+            long totalProcessedRows = 0;
+            long totalProcessedBytes = 0;
+            long totalElapsedWallTimeMills = 0;
+            for (PrestoSparkShuffleStats stats : statsList) {
+                totalProcessedRows += stats.getProcessedRows();
+                totalProcessedBytes += stats.getProcessedBytes();
+                totalElapsedWallTimeMills += stats.getElapsedWallTimeMills();
+            }
+            long totalElapsedWallTimeSeconds = totalElapsedWallTimeMills / 1000;
+            long rowsPerSecond = totalProcessedRows;
+            long bytesPerSecond = totalProcessedBytes;
+            if (totalElapsedWallTimeSeconds > 0) {
```

Comment:
what about just do something like 
```
rowsPerSecond = (long) (totalProcessedRows * 1000.0 / totalElapsedWallTimeMills)
bytesPerSecond = ...
```
?

Reply:
That's just a matter of taste I believe.

Personally I find messages like these confusing:

`Total rows 1, rows per second: 1_000_000`.

I would rather prefer to see something like `Total rows 1, rows per second: 1` (although this isn't very mathematically correct). 

What do you think?